schemas/collector-report.schema.json

```diff
@@ -0,0 +1,75 @@
+{
+  "$schema": "http://json-schema.org/schema",
+
+  "definitions": {
+    "exposure": {
+      "type": "string",
+      "enum": ["ServiceWorker", "SharedWorker", "Window", "Worker"]
+    },
+
+    "feature_name": {
+      "type": "string",
+      "pattern": "^[a-zA-Z_0-9-$@.]*$",
+      "description": "Dot separated identifier path.\n\n@example \"javascript.builtins.Array.at\"",
+      "examples": ["javascript.builtins.Array.at"]
+    },
+
+    "collector_result": {
+      "oneOf": [
+        {
+          "type": "object",
+          "properties": {
+            "exposure": { "$ref": "#/definitions/exposure" },
+            "name": { "$ref": "#/definitions/feature_name" },
+            "result": { "type": "boolean" }
+          },
+          "additionalProperties": false,
+          "required": ["exposure", "name", "result"]
+        },
+        {
+          "type": "object",
+          "properties": {
+            "exposure": { "$ref": "#/definitions/exposure" },
+            "message": {
+              "type": "string",
+              "pattern": "^threw .*$|^Testing .* in workers is not yet implemented$|^Browser does not support detection methods$",
+              "description": "Reason a result could not be collected.\n\n@example \"threw TypeError: Failed to construct 'Notification': Illegal constructor.\"\n@example \"Testing CanvasRenderingContext2D in workers is not yet implemented\"\n@example \"Browser does not support detection methods\""
+            },
+            "name": { "$ref": "#/definitions/feature_name" },
+            "result": { "type": "null" }
+          },
+          "additionalProperties": false,
+          "required": ["exposure", "message", "name", "result"]
+        }
+      ]
+    }
+  },
```

Comment:
A message may be present on _any_ result, not just `null` results.  For example, the `testConstructor()` function will capture the message of the error that was caught when trying to use a constructor with no arguments, so that we may be able to determine why the collector considered the constructor to be supported or not.

Reply:
Are there `null` results without `message`?

I'll add `message` as not required to `boolean` results. Should `message` be required by `null` results?

Reply:
I can't think of a reason that `null` would be without a `message`, so I'd consider it to be required for `null`!